build/assert-publish-aot-testApp-warning-count.ps1

```diff
@@ -0,0 +1,36 @@
+Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass
+
+$trimmingWarningCount = 0
+$aotWarningCount = 0
+foreach($line in Get-Content .\build\publishAotTestAppOutput.log) 
+{
+    Write-Host $line
+    if ($line -like "*Trim analysis warning IL*") 
+    { 
+        $trimmingWarningCount += 1
+    } elseif ($line -like "*AOT analysis warning IL*") 
+    {
+        $aotWarningCount += 1
+    }
+}
+
+$expectedTrimmingWarningCount = 11
+$expectedaotWarningCount = 19

```

Comment:
I don't think it makes much sense to track the different warnings separately. Additionally we could potentially miss other warnings (like single-file related warnings which also prevent AOT apps from working).
We should simply count all warnings - the publish of the test app should be warning free.

Reply:
> the publish of the test app should be warning free.

Unfortunately, the publish test app also includes the warnings from the "product" assemblies (as far as I can tell).

Reply:
If that's the case - could we maybe detect all "IL####" warnings instead - that's a good approximation of anything even related to AOT.